### ISSUE TITLE
Kinetics search assertion error

### ISSUE THREAD

**Issue by cgrambow** [ReactionMechanismGenerator/RMG-website#165]
Trying to get the kinetics for the CH3 + O2 -> H3COO reaction on the RMG website leads to an assertion error. Go [here](http://rmg.mit.edu/database/kinetics/results/reactant1=multiplicity%202%0D%0A1%20C%20u1%20p0%20c0%20%7B2,S%7D%20%7B3,S%7D%20%7B4,S%7D%0D%0A2%20H%20u0%20p0%20c0%20%7B1,S%7D%0D%0A3%20H%20u0%20p0%20c0%20%7B1,S%7D%0D%0A4%20H%20u0%20p0%20c0%20%7B1,S%7D%0D%0A__reactant2=multiplicity%203%0D%0A1%20O%20u1%20p2%20c0%20%7B2,S%7D%0D%0A2%20O%20u1%20p2%20c0%20%7B1,S%7D%0D%0A__res=True) and click the third reaction.

**Comment by mliu49** [comment]
Here's the traceback:
```
Traceback (most recent call last):
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/www/RMG-website/rmgweb/database/views.py", line 2307, in kineticsData
    context_instance=RequestContext(request))
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/shortcuts.py", line 45, in render_to_response
    using=using)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/loader.py", line 115, in render_to_string
    template_name, context, context_instance, dirs, dictionary)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/engine.py", line 221, in render_to_string
    return t.render(context_instance)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 209, in render
    return self._render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 201, in _render
    return self.nodelist.render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 917, in render_node
    return node.render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/loader_tags.py", line 135, in render
    return compiled_parent._render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 201, in _render
    return self.nodelist.render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 917, in render_node
    return node.render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 917, in render_node
    return node.render(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/defaulttags.py", line 224, in render
    nodelist.append(node.render(context))
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 957, in render
    output = self.filter_expression.resolve(context)
  File "/opt/anaconda/envs/rmg_env/lib/python2.7/site-packages/django/template/base.py", line 674, in resolve
    new_obj = func(obj, *arg_vals)
  File "/var/www/RMG-website/rmgweb/main/templatetags/render_kinetics.py", line 463, in get_rate_coefficients
    klist.append(kinetics.getRateCoefficient(T,P) * kfactor)
  File "rmgpy/kinetics/arrhenius.pyx", line 751, in rmgpy.kinetics.arrhenius.MultiPDepArrhenius.getRateCoefficient
    cpdef double getRateCoefficient(self, double T, double P=0.0) except -1:
  File "rmgpy/kinetics/arrhenius.pyx", line 768, in rmgpy.kinetics.arrhenius.MultiPDepArrhenius.getRateCoefficient
    assert Plist1.shape[0] == Plist2.shape[0] AssertionError
```

**Comment by cgrambow** [comment]
So presumably one of the kinetics entries for that reaction has incongruent `Plist`s?

**Comment by cgrambow** [comment]
Seems like the offending entry is index 81 in `Klippenstein_Glarborg2016`. The second `PDepArrhenius` is missing kinetics for 1 and 10 atm.

**Comment by cgrambow** [comment]
So I feel like that means that whenever someone uses that library and RMG tries to get kinetics for that reaction, it will cause an error. Seems like an issue. Is there any reason why RMG should require that identical Plists are available?

**Comment by alongd** [comment]
When MultiPDepArr calcs k
k = klow * 10**(log10(P/Plow)/log10(Phigh/Plow)*log10(khigh/klow))
It uses a collective Plow, so it seems like both Plists need to be identical, or at least have the same Plow

**Comment by cgrambow** [comment]
I'm not an expert on all the different ways that RMG evaluates rate coefficients, but presumably if we only have one `PDepArrhenius` that's valid at one of the pressures, then it should just use that to evaluate the rate coefficient if we're close to that pressure instead of summing up the rate coefficients like it does for pressures where multiple `PDepArrhenius` expressions are available. Although I'm not sure how we would handle boundary cases, e.g., in the above mentioned `Klippenstein_Glarborg2016` rate, if we're at like 15 atm, should we add in a contribution from the `Arrhenius` at 20 atm of the second `PDepArrhenius` or only interpolate between 10 and 20 atm using the first `PDepArrhenius`? I tried looking up how Chemkin would handle an expression like that, but couldn't find anything. Maybe someone with more experience in how to evaluate rate constant expressions wants to fix this?

**Comment by alongd** [comment]
I don't think the Plists indicate the pressures where each expression is valid, seems to me this is a case of lack of data - they reported the available rates, but both routes should still occur. (Of course, the farther we go from the Plist boundaries, the higher the uncertainty is.) If I understand correctly, we could solve this by changing the MultiPDepArr function to call PDepArr several times and sum up the returned values.

**Comment by alongd** [comment]
OK. Let's check if it's technically possible to calculate a PDepArr outside of it's PList (I think Chemkin complaint in situations like this).

Alternatively, we could change the format to two duplicate PDepArr. Could be simpler.

For reference, the original reactions from that publication are:
```
CH3+O2=CH3OO                   5.0E22    -3.85    2000
  PLOG/ 1.0                              5.0E22   -3.85    2000/
  PLOG/10.0                             3.35E21  -3.20    2300/
  PLOG/20.0                             3.25E29  -5.6     6850/
  PLOG/50.0                             2.83E18  -2.20    1400/
  PLOG/100.0                           1.05E19  -2.30    1800/
duplicate
CH3+O2=CH3OO                   5.0E22    -3.85    2000
  PLOG/20.0                             4.10E20  -2.94    1900/
  PLOG/50.0                             5.60E28  -5.25    6850/
  PLOG/100.0                           4.10E30  -5.7     8750/
duplicate
! Fernandes RX Luther K Troe J JPCA 110:4442-4449 2006 (fit to data)
```

**Comment by mjohnson541** [comment]
I just found some occurances of this error running RMG with the CurranPentane library.

**Comment by mjohnson541** [comment]
Interestingly, when I ran it on an older branch (few weeks ago) it doesn't have that problem.

**Comment by mliu49** [comment]
Fixed by #1341
